### PR TITLE
Madeline dev

### PR DESCRIPTION
**Pull request by mcuozzo19**
@Yojairo check JSON format if this is how you need it

### CODE REVIEW

File: website/src/test/java/com/google/sps/servlets/MockDataServletTest.java [hunk 1]
```diff
@@ -57,28 +64,29 @@ public MockDataServletTest(){
     
     public MockDataServletTester(){
     
-     int cntr = 0;
+    Datastore datastore = mock(DatastoreService.class);
```
Comment (PaulLockett) [inline website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: remove "Datastore" you are reassigning not making a new variable.

File: website/src/test/java/com/google/sps/servlets/MockDataServletTest.java [hunk 2]
```diff
@@ -57,28 +64,29 @@ public MockDataServletTest(){
...
-    try{
+    
       HttpServletRequest request = mock(HttpServletRequest.class)       ;
       HttpServletResponse response = mock(HttpServletResponse.class);    
+    
 
-      when(datastore.prepare(query).thenReturn(MockDatastoreList));
-
+      when(datastore.prepare()).thenReturn(results);
+      when(results.asIterable()).thenReturn(MockDatastoreList);//asiterable return list
```
Comment (PaulLockett) [inline website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: Define the mock results. like you define the mock of datastore. only put the line in this function.

Comment (PaulLockett) [reply website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: you should have 3 mocks being made in this function request, response, and results.

Comment (mcuozzo19) [reply website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: Okay so now results works but it still says datastore is an unknown variable

Comment (PaulLockett) [reply website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: Check my other comment. I believe that will resolve your issue.

Comment (mcuozzo19) [reply website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: I removed the "Datastore" and got the same error but I can keep messing with it

Comment (mcuozzo19) [reply website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: It is being declared in MockDataServletTester() constructor so its saying it is unfound in the MockDataServletTest class

**Comment by PaulLockett** [comment]
Remove the commented out code

File: website/src/main/java/com/google/sps/servlets/MockDataServlet.java [hunk 3]
```diff
@@ -54,15 +56,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
     Gson gson = new Gson();
     ArrayList<String> restaurants = new ArrayList<>();
+    int cntr=0;
     for (Entity entity : results.asIterable()) {
       long id = (long)entity.getProperty("idNum");
       long orderVolume = (long) entity.getProperty("openOrderVolume");
-      String message = id + " : " + orderVolume;      
+      String message = "orderVolume" + " : " + orderVolume;      
       restaurants.add(message); 
+      cntr++;
```
Comment (PaulLockett) [inline website/src/main/java/com/google/sps/servlets/MockDataServlet.java]: What is this counter for?

File: website/src/main/java/com/google/sps/servlets/MockDataServlet.java [hunk 4]
```diff
@@ -15,6 +15,7 @@
 package com.google.sps.servlets;
 
 import java.util.Random;
+import java. util. Collections;
```
Comment (PaulLockett) [inline website/src/main/java/com/google/sps/servlets/MockDataServlet.java:18]: Be sure to remove any unnecessary imports.

File: website/src/test/java/com/google/sps/servlets/MockDataServletTest.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java. util. Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.atLeast;
```
Comment (PaulLockett) [inline website/src/test/java/com/google/sps/servlets/MockDataServletTest.java]: Try and remove unnecessary dependencies.

**Review by PaulLockett — CHANGES_REQUESTED** [review]
Last clean up before merging

File: website/src/main/java/com/google/sps/servlets/MockDataServlet.java [hunk 6]
```diff
@@ -15,6 +15,7 @@
...
 import com.google.appengine.api.datastore.FetchOptions;
 import com.google.appengine.api.datastore.Key;
```
Comment (PaulLockett) [inline website/src/main/java/com/google/sps/servlets/MockDataServlet.java:20]: What are you using Key for?

File: website/src/test/java/com/google/sps/servlets/MockDataServletTest.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java. util. Collections;
```
Comment (PaulLockett) [inline website/src/test/java/com/google/sps/servlets/MockDataServletTest.java:19]: Where is Collections used in your code?

**Review by PaulLockett — APPROVED** [review]
Nit, some extra dependencies can be removed.

**Pull request merged**